crates/dojo-lang/src/scarb_internal/mod.rs

```diff
@@ -24,21 +24,36 @@ use crate::plugin::dojo_plugin_suite;
 
 pub(crate) const LOG_TARGET: &str = "dojo_lang::scarb_internal";
 
+/// Compilation information of all the units found in the workspace.
+#[derive(Debug, Default)]
 pub struct CompileInfo {
+    /// The name of the profile used to compile.
     pub profile_name: String,
+    /// The path to the manifest file.
     pub manifest_path: Utf8PathBuf,
+    /// The path to the target directory.
     pub target_dir: Utf8PathBuf,
+    /// The name of the root package.
     pub root_package_name: Option<String>,
+    /// The list of units that failed to compile.
+    pub compile_error_units: Vec<String>,
 }
 
-pub fn crates_config_for_compilation_unit(unit: &CompilationUnit) -> AllCratesConfig {
+pub fn crates_config_for_compilation_unit(unit: &CairoCompilationUnit) -> AllCratesConfig {
     let crates_config: OrderedHashMap<SmolStr, CrateSettings> = unit
-        .components
+        .components()
         .iter()
         .map(|component| {
             (
                 component.cairo_package_name(),
-                CrateSettings { edition: component.package.manifest.edition, ..Default::default() },
+                CrateSettings {
+                    edition: component.package.manifest.edition,
+                    experimental_features: ExperimentalFeaturesConfig {
+                        negative_impls: true,
+                        coupons: true,
+                    },
```

Comment:
I didn't manage to have core lib compiling with this scarb version, without enabling them. That was weird to me as experimental shouldn't be required. But as mentioned to scarb team, I need them to be enabled.

I will try to disable them again, and I should have a sync with scarb guys next week on that to see how this may impact the implementation of smart contracts.

But mainly, without those flags, Corelib doesn't compile.

Reply:
Fixed in fe86cd6c0302142ac1015b9681d49273037c0814.